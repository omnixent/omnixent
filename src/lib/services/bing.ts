import { ServiceArgs } from "./index";
import { normalizeServiceArgs } from "../utils";

const endpoint = "https://api.bing.com/osjson.aspx";

export function formatURI(options: ServiceArgs): string {
  const { t, c, l } = normalizeServiceArgs(options);
  return `${endpoint}?query=${t}&mkt=${l}-${c}`;
}

export function extractBody(response: string) {
  return JSON
    .parse(response)[1]
    ?.map((phrase: string) => phrase);
}
