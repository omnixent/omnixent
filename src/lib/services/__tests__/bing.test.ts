import { formatURI, extractBody } from '../bing';

test('Testing google formatURI', () => {
  expect(formatURI({ term: 'Node.js', language: 'it', country: 'us' })).toMatchInlineSnapshot(
    `"https://api.bing.com/osjson.aspx?query=node.js&mkt=IT-us"`,
  );
  expect(
    formatURI({ term: 'blue jeans brands', language: 'en', country: 'de' }),
  ).toMatchInlineSnapshot(
    `"https://api.bing.com/osjson.aspx?query=blue%20jeans%20brands&mkt=EN-de"`,
  );
});

test('Testing google extractBody', () => {
  expect(
    extractBody(
      `["node.js",["node.js","node.js download","node.js update","node.js fs","node.js express","node.js cos\u0027e","node.js example","node.js mysql","node.js version","node.js hello world","node.js raspberry pi","node.js server-side javascript"],[],[],{"google:suggestrelevance":[1300,1299,1298,1297,1296,1295,1294,1293,1292,1291,1290,1289]}]`,
    ),
  ).toMatchSnapshot();
  expect(
    extractBody(
      `["blue jeans brands",["blue jeans brands","blue jeans brands list","blue jeans brand names","blue jean brands for men","blue jean brands for women","blue jean brands of the 1960\u0027s and 70\u0027s"],[],[],{"google:suggestrelevance":[1300,1299,1298,1297,1296,1295]}]`,
    ),
  ).toMatchSnapshot();
});
