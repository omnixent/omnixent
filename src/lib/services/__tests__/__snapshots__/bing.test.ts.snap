// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing google extractBody 1`] = `
Array [
  "node.js",
  "node.js download",
  "node.js update",
  "node.js fs",
  "node.js express",
  "node.js cos'e",
  "node.js example",
  "node.js mysql",
  "node.js version",
  "node.js hello world",
  "node.js raspberry pi",
  "node.js server-side javascript",
]
`;

exports[`Testing google extractBody 2`] = `
Array [
  "blue jeans brands",
  "blue jeans brands list",
  "blue jeans brand names",
  "blue jean brands for men",
  "blue jean brands for women",
  "blue jean brands of the 1960's and 70's",
]
`;
